### ISSUE TITLE
To-do list

### ISSUE THREAD

**Issue by qweytr** [PhelanBavaria/ancienttimeline#50]
Here's a list of things that need to be done for the mod to get finished.
1. Update to the newest version of the game
2. New events for the new religions
3. Are the discoveries still done?
4. Province names should be changed to something more contemporary. (Mostly undone still afaik)
5. History fixes. For now all I can say that the Scordisci are missing. There are likely a lot more mistakes, so you could go though historical maps and compare them to the AT start setup on that date.

**Comment by Firesoul7** [comment]
I can't even face province discoveries, my personal assessment is that the Greeks would be aware of most coastal provinces, or at least all major ports as far as the pillars of Hercules, but no further, and they would also be extremely unknowledgable about any inland territories that aren't inhabited by Greeks. As for the Achaemenids, they would be aware of northern Greece and parts of Northern India, but otherwise very little beyond their own borders.

So, besides this, which parts of the to-do list are my responsibility?

**Comment by qweytr** [comment]
The areas of responsibility are pretty much gone at this point. Anyone can go and do anything they want, but do let the me know what you are going to do so no two people end up doing the same thing.

**Comment by Firesoul7** [comment]
I can deal with the animism first

EDIT: This was done

**Comment by PanzerfaustJoe** [comment]
It is a shame that pretty much everyone decided to leave and do whatever else. Especially because I can only work on this mod every other day since I have life stuff going on.

Speaking of, I will have Dacia and new province histories done later tonight, and maybe some national ideas.

**Comment by Firesoul7** [comment]
I've done something incredibly stupid, I decided it was probably best to just copy across the MOD file in case there were changes in order to make it compatible with the latest version, but I forgot that my mod files are in a weird place (weird but I like it) so there's a special line to tell the MOD file where my mod files are, and even stupider I forgot to create a backup of this line. I also can't remember how the line works. I'm editing path="mod/ancienttimeline" so it goes from C:\Users\Roman\Documents\Paradox Interactive\Europa Universalis IV\mod\ancienttimeline to C:\Users\Roman\Documents\Paradox Interactive\Modding\ancienttimeline\ancienttimeline\ancienttimeline could someone tell me what the line should read please? Thank you

**Comment by qweytr** [comment]
I don't know how to go up directories with the mod path. The command is usually .. so my guess is "..\Modding\ancienttimeline\ancienttimeline\ancienttimeline"

**Comment by Firesoul7** [comment]
That's the easy part, it's the dots I'm struggling with

**Comment by Firesoul7** [comment]
Thank you, it works

**Comment by PanzerfaustJoe** [comment]
5 and 6 are done

**Comment by Firesoul7** [comment]
I believe that Asturias should be culturally Gallaic and not Celtiberian since all their provinces are Gallaic, although regarding that, many provinces in Spain are displayed as having 'No Culture' (such as Leon) despite their files seeming fine, could someone have a look at those thanks?

**Comment by Firesoul7** [comment]
https://en.wikipedia.org/wiki/Yelang#Nanyue
Note that according to this Yelang is Yi culture, not Miao culture. They should therefore probably be changed back to animism, their culture changed to Yi, and maybe the Miao tag added to AT. Yelang already owns a Yi province.

EDIT: In fact I'll change it myself, if you disagree discuss it on here

EDITEDIT: The Yi people now have their own religion.

**Comment by Firesoul7** [comment]
The only animism left in lands with countries is in Hispania, and I'm not so comfortable working there until the issues in the region are fixed. I'll probably start working on my backlog of ET idea groups at least for today.

**Comment by qweytr** [comment]
We can consider the religions finished for the first release.

**Comment by Science-Recon** [comment]
Do we have icons for the religions now, then? Or is the artwork going to be done together at the end?

**Comment by Firesoul7** [comment]
We still don't have the icons, no.

**Comment by qweytr** [comment]
The religion icons are indeed still to be done. You can do that if you want. I don't think there are going to be any more religions added before the first release.

**Comment by Firesoul7** [comment]
By you do you mean me or him?

**Comment by Science-Recon** [comment]
Well, I'm not a very good graphical artist, but I can sort it out at get it working if I'm given the icons.

**Comment by Firesoul7** [comment]
I'm not exactly a great graphical artist either, Qweytr, do you think you
could persuade Adtriarior to do them for us? Judging by his event pictures
I'd say he's the best person for the job.

**Comment by qweytr** [comment]
I'll do the religion icons at some point if no one else wants to.

**Comment by Firesoul7** [comment]
Did you do the ET icons? If so it would be great if you did these too

**Comment by Firesoul7** [comment]
I've done a little experiment on the current implementation of Macedon/Achaemenids, here are my results:
Trial 1:
War never begins, Macedon dies slow death to rebels
Trial 2:
War begins, Macedon marches round in circles with their army in Achaemenids before sieging and taking Babylon along with Tikrit, Wasit and Hillah. Meanwhile the Achaemenids have free reign in Greece and take 5 provinces although they later collapse to rebels
Trial 3:
Most successful so far, Macedon takes Kyzikos and loses no land. They get very high war score but demand only Cyprus when the war ends. Most of the war they hunt Achaemenid stacks, Achaemenids begin to collapse to rebels during war. Collapse effects about a year after war
Trial 4:
Theoretically successful for Macedon, but weird. First they seized Kyzikos, but then charged right for Babylon and took it. The Achaemenids despite stationing a large army next to empty Kyzikos made no attempt to retake it, and Greece remained secure. The Achaemenids and Macedon were both taken over by rebels, and Macedon seized some provinces here and there from rebels and Achaemenids, until both armies were destroyed utterly by rebels. Finally the two nations collapsed simultaneously
Trial 5: Macedon takes Astacus and Kyzikos. Earlier they take Helicore but the Achaemenids retake it. Rebels overpower both nations during the war and the Achaemenids collapse within days of the war ending

What I believe these results show: that the Achaemenids have too many troops. They should have decreased forcelimits. Even though they lose every battle against Alexander they also outflank him every time. The other thing these results show is that the rebels are too much for anyone. I think their current numbers and strength is fine, but they appear too early. War exhaustion needs to be greatly slowed. I also think that while big rebellions are done excellently, every rebellion I see in this is huge, which is not historical. Many rebellions were a mere nuisance, although of course there were larger rebellions. The average result should be Alexander taking all land as far as and including Egypt. This never happens.

**Comment by qweytr** [comment]
The most straightforward way to reduce the power of the Achemenids is to reduce the number of states they have. They start out as an empire, which gives them 10 additional states. We could either reduce the number of states empires get or make them into a kingdom.

**Comment by Firesoul7** [comment]
They shouldn't be a Kingdom, and we want all empires to be fairly unstable
in this period, so I recommend reducing the number of states they get

**Comment by PanzerfaustJoe** [comment]
I agree, but I dont want them to be unplayable. I think the Achaemenids should a chance, no matter how slim, to survive Alexander and/or rebels. Having the same thing happen every game makes it boring. 

For example, the collapsing Roman Empire in ET happens pretty much every game, WAY before the barbarian migrations.

**Comment by Firesoul7** [comment]
From my experience Rome by no means collapses every game before the
barbarians arrive, although as what I would consider as a half finished
mechanic it still needs work. Rebels seem to kill everyone when countries
are at war in AT, so I do think war exhaustion should be slowed, which is
part of what you're suggesting. If you read my trials you'll see that so
far Alexander basically isn't a problem for the Achaemenids (which is bad)
while rebels are (which is worse).

**Comment by PanzerfaustJoe** [comment]
I just played an ET game yesterday, and Rome collapsed yet again from rebels. I don't doubt what you say but I think if the chance of Rome collapsing was decreased about 20% more it would be perfect. But enough about ET.

**Comment by PhelanBavaria** [comment]
Maybe we can give nations an "at war" modifier which makes them more stable.
Historically this is realistic, because who is there to rise up if they are all in the army?

**Comment by Firesoul7** [comment]
I'm not sure, although it's true that armies at the time were basically all
levied peasantry, practical concerns (like working the fields and actually
conscripting people) meant that a lot of people stayed behind, and with the
army away and the king with them it was often a fairly good opportunity to
rise up. I do think that current war exhaustion is insanely OP though.

**Comment by qweytr** [comment]
The reason that the revolts are currently too powerful is probably that rebels get higher morale with higher technology. That value is still unchanged from ET even though the tech levels in AT are much higher, thus causing the rebels to have much higher morale than they should have.

**Comment by Firesoul7** [comment]
Does that have an easy fix?

**Comment by Science-Recon** [comment]
Is it not in the defines?

**Comment by qweytr** [comment]
It's a single line in the defines, NDefines.NCountry.REVOLT_TECH_MORALE The current value is clearly too high, but I'm not entirely sure what it should be changed to.

**Comment by Science-Recon** [comment]
Time for some testing then, I guess.

**Comment by qweytr** [comment]
I have now decreased the value a bit. That change is not based on any testing, so if you think it should be higher or lower feel free to change it.

**Comment by Firesoul7** [comment]
I guess I can do some more trials at some point

**Comment by Firesoul7** [comment]
Regarding the OP, I've edited a couple of names to make them more historical. The provinces I edited were the glitch ones in Spain that for some reason continue to show as no culture. This glitch is still completely baffling to me, but it should be fixed before initial release. Also, no provinces in Germania except the Prussian ones have culture. This is because the cultures in the file that lists cultures has been changed, but the provinces and countries themselves haven't been edited to reflect this. This will also need to be fixed for initial release. Also, it's a little weird that the Gepids live in Prussian territory, I don't know if the extent of Prussian territory is wrong or if the Gepids are in the wrong place, whichever it is this should be rectified.

I think I'll add two religions to Spain eventually. I studied Spanish history for a month in Spain, and it seems that this is necessary from what I learned. I'm not comfortable with adding these religions while the provinces are glitched though.

**Comment by qweytr** [comment]
I've added the German cultures to the OP. I hope the validator can find the Spanish culture bug, but for now there are so many errors that it's hard to find the important ones among all the others. I might soon start working on fixing the errors to reduce the number of reports the validator gives.

**Comment by Firesoul7** [comment]
Less important but the same glitch exists in two provinces in Japan. It's simply bizarre. Could it be that there's too many provinces? We don't need so many provinces in wastelands after all. 

EDIT: For anyone reading this later, it turned out to be an issue with encoding, it's fixed now.

**Comment by qweytr** [comment]
There are less provinces than in ET and it's true that we don't need that many provinces for the wastelands, but by keeping them it will be easier to make them colonizable in later updates if we want to.

**Comment by Firesoul7** [comment]
True. Do you think it could be an issue with the number of provinces that
we're suffering?

**Comment by qweytr** [comment]
No. The number of provinces can't have anything to do with it.

**Comment by Science-Recon** [comment]
Who is going to add the institutions? I can code them in and give them temporary icons (that way they can be switched once there's art for it), but if anyone else wants to do it that's fine. Also, consorts and personalities are something we can add (though not necessarily with everyone) we may also want to change/ add personalities and traits. The Coptic mechanics are cool. Does anyone know how to remove their interface (the one in religion view, not the HRE-type one) or at least send it to the bottom so that we can maybe add this in addition to the religion's main mechanic (E.g. Personal deities)? I've tried but not succeeded. Final note: could we change the Hellenism icon to the Hellenism icon used by the converter?

**Comment by qweytr** [comment]
I have now started adding the institutions and my plan is to finish it myself as well. Feel free to work on any of those other points you mentioned. But religions should only have one mechanic each, i.e. holy sites or deities, not both.

**Comment by Science-Recon** [comment]
Ok. I thought that the holy sites mechanic would be a good feature for all religions, similar to how there are holy sites in CKII. They'd probably only give minor bonuses. However, I have yet to make the interface cooperate, so I'll leave it for now.

**Comment by Firesoul7** [comment]
The first post of this should either be updated, or this whole issue closed and another one opened

**Comment by Science-Recon** [comment]
Yea, I think an edit of the OP should suffice.

**Comment by Firesoul7** [comment]
No problem at all, the OP can be edited, but do we know what there actually is to do? What I'm aware of: events for a bunch of the religions, if you want specifics as to which religions then I can give you the list I have; maybe discoveries, but these might be done; maybe some events for like Rome taking places over, inheritance of Pergamon, Maurya Empire, stuff like that; Ages; the missing history files

**Comment by qweytr** [comment]
I have now edited the first post.

**Comment by Firesoul7** [comment]
In my opinion, we shouldn't look into initial release until we have at least one (ideally all) of the 'wow' elements that would instantly make us recognised as better than Imperium. These are:
Converter between AT and ET (big)
Province Abandonment (small, i.e. needs something else)
Americas (medium to big)

Additionally, regarding the history, https://en.wikipedia.org/wiki/Pyu_city-states Pyu needs to be added to the later dates. Finally, I believe that we should fill in Japan. My impression is that the setup would be all but identical to that in the 51 bookmark on ET. Emishi could be given the whole Yamato land on Honshu before Emperor Jimmu, with Yamato ruling Kyushu, but this is anyway far before the start date we currently begin at

**Comment by Science-Recon** [comment]
I would be inclined to agree. At this point we're not going to be first so we should be the best and there's no point trying to get this out quickly and rushing things.

**Comment by Science-Recon** [comment]
How do we plan on doing the converter and would it be worth creating an issue for that separate to this?

**Comment by Firesoul7** [comment]
I'll create such an issue now

**Comment by Science-Recon** [comment]
I'll probably have some time at the weekend and although I'll be away from my PC I should have my laptop with me so I may go through and add/change/update localisations.

**Comment by Firesoul7** [comment]
I've just finished the last of the missing history files. Tomorrow I'll try to add events for another religion, and I'll keep going until all the religions are done.
